weatherman.py

```diff
@@ -0,0 +1,164 @@
+import os
+import argparse
+from datetime import datetime
+
+
+RED = "\033[31m"
+BLUE = "\033[34m"
+RESET = "\033[0m"
+
+
+class WeatherReading:
+    def __init__(self, date, max_temp, min_temp, max_humidity, mean_humidity):
+        self.date = date
+        self.max_temp = max_temp
+        self.min_temp = min_temp
+        self.max_humidity = max_humidity
+        self.mean_humidity = mean_humidity
+
+    def __repr__(self):
+        return (f"WeatherReading(date={self.date}, max_temp={self.max_temp}, "
+                f"min_temp={self.min_temp}, max_humidity={self.max_humidity}, "
+                f"mean_humidity={self.mean_humidity})")
+
+
+def parse_weather_file(file_path):
+    readings = []
+    try:
+        with open(file_path, 'r') as file:
+            lines = file.readlines()[1:]  
+            for line in lines:
+                parts = line.strip().split(',')
+                if len(parts) < 9 or '' in (parts[1], parts[3], parts[7], parts[8]):
+                    continue  
+                date = parts[0]
+                max_temp = int(parts[1])
+                min_temp = int(parts[3])
+                max_humidity = int(parts[7])
+                mean_humidity = int(parts[8])
+                reading = WeatherReading(date, max_temp, min_temp, max_humidity, mean_humidity)
+                readings.append(reading)
+    except FileNotFoundError:
+        print(f"File not found: {file_path}")
+    return readings
+
+
+class WeatherCalculations:
+    def __init__(self):
+        self.highest_temp = None
+        self.lowest_temp = None
+        self.highest_humidity = None
+        self.total_max_temp = 0
+        self.total_min_temp = 0
+        self.total_mean_humidity = 0
+        self.num_readings = 0
```

Comment:
Why are we supposed to define base values to 0?

Reply:
It is because we are storing sum of different values in these variables for calculating averages. Initialization from zero ensures that sum is calculated correctly.

Reply:
Can't we calculate sum on the fly and update class level variable directly from there?